addons/mail/static/src/discuss/call/common/rtc_session_model.js

```diff
@@ -1,16 +1,40 @@
 import { Record } from "@mail/core/common/record";
+import { Deferred } from "@web/core/utils/concurrency";
+
+const ASYNC_GETTER_TIMEOUT = 120_000;
 
 export class RtcSession extends Record {
     static id = "id";
     /** @type {Object.<number, import("models").RtcSession>} */
     static records = {};
+    static awaitedRecords = new Map();
     /** @returns {import("models").RtcSession} */
     static get(data) {
         return super.get(data);
     }
-    /** @returns {import("models").RtcSession|import("models").RtcSession[]} */
-    static insert(data) {
-        return super.insert(...arguments);
+    /** @returns {Promies<import("models").RtcSession>} */
+    static async getWhenReady(id) {
+        const session = this.get(id);
+        if (!session) {
+            let deferred = this.awaitedRecords.get(id);
+            if (!deferred) {
+                deferred = new Deferred();
+                this.awaitedRecords.set(id, deferred);
+                setTimeout(() => {
+                    deferred.resolve?.();
+                    this.awaitedRecords.delete(id);
+                }, ASYNC_GETTER_TIMEOUT);
+            }
+            return deferred;
+        }
+        return session;
+    }
+    /** @returns {import("models").RtcSession} */
+    static new() {
+        const record = super.new(...arguments);
+        this.awaitedRecords.get(record.id)?.resolve?.(record);
+        this.awaitedRecords.delete(record.id);
```

Comment:
Ideally this should clear the timeout too (though it won't hurt to actually call resolve again, I think it does nothing).

Reply:
@seb-odoo yes I thought managing the clear timeout would bloat the code for a minor benefit